automatic/poi/update.ps1

```diff
@@ -0,0 +1,56 @@
+import-module au
+
+$releases = 'https://github.com/poooi/poi/releases'
+
+function global:au_BeforeUpdate { Get-RemoteFiles -Purge -NoSuffix }
+
+function global:au_SearchReplace {
+    @{
+        ".\tools\chocolateyInstall.ps1" = @{
+            "(?i)(^\s*file\s*=\s*`"[$]toolsDir\\).*" = "`$1$($Latest.FileName32)`""
+        }
+        "$($Latest.PackageName).nuspec" = @{
+            "(\<releaseNotes\>).*?(\</releaseNotes\>)" = "`${1}$($Latest.ReleaseNotes)`$2"
+        }
+        ".\legal\VERIFICATION.txt" = @{
+            "(?i)(1\..+)\<.*\>"         = "`${1}<$($Latest.URL32)>"
+            "(?i)(checksum type:\s+).*" = "`${1}$($Latest.ChecksumType32)"
+            "(?i)(checksum:\s+).*"      = "`${1}$($Latest.Checksum32)"
+        }
+    }
+}
+
+function global:au_GetLatest {
+
+    $stableReleaseFound = $false
+    $url = ''
+    $lastVersionString = ''
+    do {
+        $download_page = Invoke-WebRequest -Uri "$releases$lastVersionString" -UseBasicParsing
+        $urls = $download_page.links | ? href -match 'poi-setup-.+\.exe$' | % href
+        $stableUrls = @($urls) -match 'poi-setup-\d+\.\d+\.\d+\.exe$'
+        if($stableUrls.Count -gt 0) {
+            $url = $stableUrls | select -First 1
+            $stableReleaseFound = $true
+        }
+        else {
+            $lastUrl = $urls | select -Last 1
+            $lastVersionString = "?after=$(Split-Path(Split-Path($lastUrl)) -Leaf)"
+        }
+    }
+    until($stableReleaseFound)
+
+    $license = (Invoke-WebRequest -Uri "https://cdn.rawgit.com/poooi/poi/master/LICENSE" -UseBasicParsing).Content
```

Comment:
Please don't download the license like this, because of the following reasons:

1. As it is, downloading it during update provides zero benefits in its current state. If you need to download it during updating, you also need to verify that the license haven't change.
2. Having this code in the `au_GetLatest` means the license will be downloaded every time, this isn't needed. Better to have it in either `au_BeforeUpdate` or `au_AfterUpdate` which will only be called if a new version is available.
3. You're using a rawgit cdn url with a branch name. This is permanently cached the first time the url is requested, and will always provide the same file even if the file have been changed upstream.

Bottom line, unless you're willing to add verification that the license being downloaded is the same type as the one we have before (or a different license that allows distribution), please add the license itself in this PR, and remove the code to download the license.

Reply:
Fixed in f7c8b6b.